tests/loader_testing_main.cpp

```diff
@@ -48,6 +48,13 @@ int main(int argc, char** argv) {
     fs::delete_folder(fs::path(FRAMEWORK_BUILD_DIRECTORY) / "explicit_layer_manifests");
     fs::delete_folder(fs::path(FRAMEWORK_BUILD_DIRECTORY) / "implicit_layer_manifests");
 
+    // make sure the tests don't find these env-vars if they were set on the system
+    remove_env_var("VK_ICD_FILENAMES");
```

Comment:
Did `VULKAN_SDK` matter?

It might be more resilient to remove any environment variable that starts with `VK_` or `VULKAN_`.

Reply:
Hmmm These 5 environment variables are the _main_ ones to worry about. VULKAN_SDK isn't read by the loader in any which way.
XDG_STUFF & HOME are isolated because when they are used, the tests make sure that those paths go to the test directories instead of any valid path. But thinking about it, they should also just be changed to point at the right place. But that can come in a PR which also refactors how those paths are redirected, which is in the pipeline of things to change. This PR should make the tests work as expected.